TeamCode/src/main/java/org/firstinspires/ftc/teamcode/OctoQuadLocalizer.java

```diff
@@ -0,0 +1,163 @@
+        increase when you move the robot forward. And the Y (strafe) pod should increase when
+        you move the robot to the left.
+         */
+        public OctoQuadFWv3.EncoderDirection xDirection = OctoQuadFWv3.EncoderDirection.FORWARD;
+        public OctoQuadFWv3.EncoderDirection yDirection = OctoQuadFWv3.EncoderDirection.REVERSE;
+
+    }
+
+    public static Params PARAMS = new Params();
+
+    public final OctoQuadFWv3 octoquad;
+    private Pose2d currentPose;
+
+    public OctoQuadLocalizer(HardwareMap hardwareMap, double inPerTick, Pose2d initialPose) {
+        FlightRecorder.write("OCTOQUAD_PARAMS", PARAMS);
+        // TODO: make sure your config has an OctoQuad device with this name
+        //   see https://ftc-docs.firstinspires.org/en/latest/hardware_and_software_configuration/configuring/index.html
+        octoquad = hardwareMap.get(OctoQuadFWv3.class, "octoquad");
+        currentPose = initialPose;
+
+        octoquad.setSingleEncoderDirection(PARAMS.odometryPortX, PARAMS.xDirection);
+        octoquad.setSingleEncoderDirection(PARAMS.odometryPortY, PARAMS.yDirection);
+
+        octoquad.setLocalizerPortX(PARAMS.odometryPortX);
+        octoquad.setLocalizerPortY(PARAMS.odometryPortY);
+
+
+
+        double mmPerTick = 25.4 * inPerTick;
+        octoquad.setLocalizerCountsPerMM_X((float) (1 / mmPerTick));
+        octoquad.setLocalizerCountsPerMM_Y((float) (1 / mmPerTick));
+
+
+        // OctoQuad offset scheme is different than Pinpoint; these are intentionally switched
+        octoquad.setLocalizerTcpOffsetMM_X((float) -(mmPerTick * PARAMS.parYTicks));
+        octoquad.setLocalizerTcpOffsetMM_Y((float) (mmPerTick * PARAMS.perpXTicks));
+
+        octoquad.setLocalizerImuHeadingScalar((float) PARAMS.angularScalar);
+
+        octoquad.setLocalizerVelocityIntervalMS(25);
+
+
+
+        // Reset the localization, IMU, and encoders
+        octoquad.resetEverything();
+
+        ElapsedTime timeout = new ElapsedTime();
+        OctoQuadFWv3.LocalizerStatus currentStatus = octoquad.getLocalizerStatus();
+        while (currentStatus != OctoQuadFWv3.LocalizerStatus.RUNNING) {
+            try {
+                Thread.sleep(10);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+            if (timeout.seconds() > 3) {
+                Log.println(
+                        Log.WARN,
+                        "OctoQuadLocalizer",
+                        "init: Calibration timeout reached, OQ still not ready. OctoQuad reports " + currentStatus
+                );
+                RobotLog.addGlobalWarningMessage(
+                        "OctoQuad still not ready, timeout reached. OctoQuad reports " + currentStatus +
+                                " Continuing anyway, good luck!"
+                );
+                break;
+            }
+            currentStatus = octoquad.getLocalizerStatus();
+        }
```

Comment:
Does this normally need to run for a while? I don't really like the idea of blocking the rest of initialization for 3s here. If this is really necessary I'd rather build a "localizer ready" signal that's used to block trajectory actions from starting (maybe just the first one?).

I'm also dubious about continuing with this being unavailable. It doesn't really make sense to run a trajectory action when there is no localization (teams concerned about possibility should make their own fallback localizer).